internal/httptransport/httptransport.go

```diff
@@ -132,6 +132,25 @@ type bodyWrapper struct {
 	tid int64
 }
 
+func (bw *bodyWrapper) Read(b []byte) (n int, err error) {
+	start := time.Now()
+	n, err = bw.ReadCloser.Read(b)
+	stop := time.Now()
+	bw.t.Handler.OnMeasurement(model.Measurement{
+		HTTPResponseBodyPart: &model.HTTPResponseBodyPartEvent{
+			// "Read reads up to len(p) bytes into p. It returns the number of
+			// bytes read (0 <= n <= len(p)) and any error encountered."
+			Data:          b[:n],
```

Comment:
Do we need to make a copy here?

Reply:
We are not making a copy. See this example code:

```Go
package main

import "fmt"

func GetBodyAndModifyIt(b []byte) {
	b[2] = 4
}

func main() {
	b := make([]byte, 7)
	fmt.Printf("slice before: %+v\n", b)
	GetBodyAndModifyIt(b[1:4])
	fmt.Printf("slice after: %+v\n", b)
}
```

and how it runs:

```
$ go build .
$ ./main 
slice before: [0 0 0 0 0 0 0]
slice after: [0 0 0 4 0 0 0]
```

and [what Dave Cheney has to say about slices](https://dave.cheney.net/2018/07/12/slices-from-the-ground-up):

> [...] Just remember that any time you assign, subslice, or pass or return, a slice, you’re making a copy of the three fields in the slice header; the pointer to the underlying array, and the current length and capacity.

So, my point was rather the converse: do we need to copy to preserve the original from further modifications? Is the current non-copying behaviour fine?

Reply:
https://github.com/ooni/netx/issues/18